Google Assistant SDK: allow personal results documentation

tronikos opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Document steps to enable personal results.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/88871
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

tronikos committed:
Update google_assistant_sdk.markdown

tronikos committed:
Update google_assistant_sdk.markdown

frenck commented on source/_integrations/google_assistant_sdk.markdown:
```diff
@@ -87,6 +87,44 @@ The integration setup will next give you instructions to enter the [Application
 
+## Enable personal results
+
+If you want to enable personal commands (e.g. "remind me tomorrow at 9pm to take out trash") follow below steps after successfully installing the integration. This is for advanced users since it requires creating an OAuth client ID of Desktop app, running a Python program on your desktop (or laptop), and copying the resulting credentials to your Home Assistant config directory.
```
```suggestion
If you want to enable personal commands (e.g. "remind me tomorrow at 9 pm to take out the trash") follow the steps below after successfully installing the integration. This is for advanced users since it requires creating an OAuth client ID of Desktop app, running a Python program on your desktop (or laptop), and copying the resulting credentials to your Home Assistant config directory.
```

frenck commented on source/_integrations/google_assistant_sdk.markdown:
```diff
@@ -87,6 +87,44 @@ The integration setup will next give you instructions to enter the [Application
+
+{% details "Create credentials file" %}
+
+1. Navigate to https://console.cloud.google.com/apis/credentials
```
Let's avoid using bare URLs

frenck commented on source/_integrations/google_assistant_sdk.markdown:
```diff
@@ -87,6 +87,44 @@ The integration setup will next give you instructions to enter the [Application
+8. On your Windows or Linux or Mac machine download Python if you don't have it already.
+9. Open terminal (on windows click on start and then type cmd).
+10. On the terminal run the following commands (preferably in a Python virtual environment): 
+11. `python -m pip install --upgrade google-auth-oauthlib[tool]`
```
I think this might be a bit overboard for the average user. How would they even do that?

tronikos replied:
Agreed. That's why the whole section starts with "This is for advanced users". I changed the heading to "Enable personal results (for advanced users)" to make it more clear.

frenck commented on source/_integrations/google_assistant_sdk.markdown:
```diff
@@ -87,6 +87,44 @@ The integration setup will next give you instructions to enter the [Application
+11. `python -m pip install --upgrade google-auth-oauthlib[tool]`
+12. Under Windows: `google-oauthlib-tool --scope https://www.googleapis.com/auth/assistant-sdk-prototype --scope https://www.googleapis.com/auth/gcm --save --client-secrets %userprofile%\Downloads\client_secret.json`
+13. Or under Linux: `google-oauthlib-tool --scope https://www.googleapis.com/auth/assistant-sdk-prototype --scope https://www.googleapis.com/auth/gcm --save --client-secrets ~/Downloads/client_secret.json`
+14. A browser window will open asking you to select account to continue to the cloud project you created earlier.
```
```suggestion
14. A browser window will open asking you to select the account to continue to the cloud project you created earlier.
```

frenck commented on source/_integrations/google_assistant_sdk.markdown:
```diff
@@ -97,9 +135,10 @@ If broadcasting doesn't work, make sure: the speakers aren't in do not disturb m
-
-Limitations of the underlying library are listed [here](https://github.com/tronikos/gassist_text#limitationsknown-issues) (media playback, routines, and personal results are not working).
+- Multiple Google accounts are not supported.
+- If you see the issued commands in https://myactivity.google.com/myactivity the integration is working fine. If the commands don't have the expected outcome don't open an issue in Home Assistant Core project. Instead open an issue in https://github.com/googlesamples/assistant-sdk-python/issues. Examples of known Google Assistant API issues:
```
Let's avoid the use of bare URLs

tronikos committed:
Apply suggestions from code review

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

tronikos committed:
Update google_assistant_sdk.markdown

tronikos committed:
Merge branch 'next' into patch-1

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.

Lash-L committed:
Fix typo in Roborock room cleaning (#27718)

wrt54g committed:
Update URL in readme (#27720)

tronikos committed:
Merge branch 'next' into patch-1

masto committed:
Fix AQI template sensor example (#27724)

It was a little unclear where to put the sample YAML because it was missing the top-level `template:`.

balloob committed:
2023.6.1 (#27721)

raman325 committed:
Add missing breaking change in 2023.6 release notes (#27728)

Co-authored-by: Franck Nijhof <git@frenck.dev>

wrt54g committed:
Update Persistent Notification documentation (#27717)

* Update URL in persistent notification

* Add My Home Assistant URLs

* Fix bug

* Fix bug (attempt 2)

hmmbob committed:
Xiaomi vacuum: add information on goto command, reorder services (#27711)

* Add information on goto command, reorder services

* Lower visibility of the note

starkillerOG committed:
Clearify Reolink intial setup (#27722)

* Clearify Reolink intial setup

* fix spelling

* Apply suggestions from code review

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update reolink.markdown

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

c0ffeeca7 committed:
Yellow: fix docs error on LED color code (#27730)

c0ffeeca7 committed:
telegram: update link to docs on creating a bot (#27729)

Drafteed committed:
Fix details block bottom margin (#27726)

joostlek committed:
Clarify need for YouTube channel (#27739)

wrt54g committed:
Update Google Wifi and Google Translate URLs (#27743)

* Remove Google Wifi URL

* Update Google translate URL

* Update SigFox URL

smarthomejunkie committed:
Update tutorial.markdown (#27752)

Added a video turorial

smarthomejunkie committed:
Update trigger.markdown (#27740)

Added video tutorial on Trigger IDs

chiefdragon committed:
Update tado.markdown (#27750)

Update documentation to reflect the ability to manually set auto-mode.

starkillerOG committed:
Add Reolink troubleshooting step (#27751)

* Add Reolink troubleshooting step

* Update reolink.markdown

* tiny tweaks

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

DominusExult committed:
Update anel_pwrctrl.markdown (#27731)

* Update anel_pwrctrl.markdown

Added notes about supported firmware, tested devices and fixed port descriptions.

* tiny tweaks

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 3.5.2 to 3.5.3 (#27757)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump sass-embedded from 1.63.2 to 1.63.3 (#27758)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

joostlek committed:
Remove duplicate configuration header (#27759)

jorik12345 committed:
Add Reolink E1 Outdoor Pro as supported (#27766)

Co-authored-by: Jorik Rovers <jr@ehvg.nl>
Co-authored-by: starkillerOG <starkiller.og@gmail.com>

elchininet committed:
Improve a statement (#27769)

Color variables should be enclosed between quotation marks, quotation marks could be single or double, but they are quotation marks, not apostrophes.

raman325 committed:
Make device parameter for area selector optional (#27770)

lymanepp committed:
Enhance tomorrowio documentation for free API keys (#27768)

https://github.com/home-assistant/core/issues/94509

ludeeus committed:
Add some clarification about network storage (#27753)

* Add some clarification about network storage

* tiny tweaks

* more tiny tweaks

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

starkillerOG committed:
Add Reolink Lumus to unsupported list (#27777)

dependabot[bot] committed:
Bump dessant/lock-threads from 4.0.0 to 4.0.1 (#27773)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

c0ffeeca7 committed:
Zone: update screenshot (#27765)

c0ffeeca7 committed:
OwnTracks integration: update docs (#27762)

c0ffeeca7 committed:
Remove changes unrelated to PR27770 (#27780)

c0ffeeca7 committed:
Platform: replace component by integration (#27778)

c0ffeeca7 committed:
Packages: replace component by integration (#27776)

c0ffeeca7 committed:
Authentication, events: replace component by integration (#27775)

c0ffeeca7 committed:
Component: replace by integration as it has been renamed (#27774)

c0ffeeca7 committed:
Alexa: replace component by integration (#27782)

c0ffeeca7 committed:
Various files: replace component by integration (#27781)

c0ffeeca7 committed:
Various files: Replace component by integration (#27783)

c0ffeeca7 committed:
Integrations starting with a-c: Replace component by integration (#27784)

bieniu committed:
Update the consumption type information in Shelly documentation (#27792)

c0ffeeca7 committed:
Logger integration: replace term component by integration (#27794)

c0ffeeca7 committed:
L-integrations: replace term component by integration (#27793)

c0ffeeca7 committed:
k-integrations: Rename component to integration (#27791)

c0ffeeca7 committed:
I-integrations: Rename component to integration (#27790)

c0ffeeca7 committed:
H-integrations: Rename component to integration (#27789)

c0ffeeca7 committed:
G-integrations: Rename component to integration (#27788)

c0ffeeca7 committed:
E-integrations: Rename component to integration (#27786)

c0ffeeca7 committed:
Rename component to integration (#27785)

c0ffeeca7 committed:
F-integrations: Rename component to integration (#27787)

Co-authored-by: Franck Nijhof <git@frenck.dev>

dependabot[bot] committed:
Bump racc from 1.7.0 to 1.7.1 (#27810)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump google-protobuf from 3.23.2 to 3.23.3 (#27811)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

c0ffeeca7 committed:
N-integrations: replace term component by integration (#27812)

c0ffeeca7 committed:
S-integrations, 01: replace term component by integration (#27818)

c0ffeeca7 committed:
S-integrations, 02: replace term component by integration (#27819)

c0ffeeca7 committed:
1-wire: remove warning that integration via SysBus is deprecated (#27813)

c0ffeeca7 committed:
P-integrations: replace term component by integration (#27815)

c0ffeeca7 committed:
r-integrations: replace term component by integration (#27817)

c0ffeeca7 committed:
P-Q-integrations: replace term component by integration (#27816)

c0ffeeca7 committed:
o-integrations: Replace term _component_ by _integration_ (#27814)

Co-authored-by: Franck Nijhof <git@frenck.dev>

c0ffeeca7 committed:
Thread: add link to HomeKit controller integration (#27820)

- rather than pointing to HomeKit, as HomeKit itself does not support Thread onboarding

by-cx committed:
Adjustment of setup of EZVIZ camera (#27799)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

spacegaier committed:
Missing blanks and tags in plant.markdown (#27827)

jpelgrom committed:
Matter: remove Google Home requirement for Android commissioning (#27824)

Lx committed:
Update light.template documentation (#27822)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Telegram: add more detailed procedure (#27826)

Co-authored-by: Franck Nijhof <git@frenck.dev>

c0ffeeca7 committed:
13d voice control: fix link to atom echo config file (#27832)

c0ffeeca7 committed:
13d voice control clarify which button to press (#27833)

dependabot[bot] committed:
Bump sass-embedded from 1.63.3 to 1.63.4 (#27834)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

balloob committed:
2023.6.2 (#27831)

c0ffeeca7 committed:
S-integrations: replace term component by integration (#27835)

c0ffeeca7 committed:
W-Z-integrations: replace term component by integration (#27838)

c0ffeeca7 committed:
t-integrations: replace term component by integration (#27836)

c0ffeeca7 committed:
U-V-integrations: replace term component by integration (#27837)

jbouwh committed:
Drop code owner imap (#27843)

ArveVM committed:
Update Packages folder - simplify (#27176)

* Update Packages folder - simplify

Replacement/simplification of point I was trying to make in #26320. Not sure what happend to that one, I might have cocked up something so it got stuck;)

have tried to simplify;
- removed the option for !include_dir_merge_named from origninal doc, so now this is all about !include_dir_named
- added info on this way of including keep indentation "as if written in configuration.yaml"

The purpose of this PR is to be able to refer to documentation when urging people to use "packages-folders with the indentation of configuration.yaml" 

Currently I've seen several examples where they have read the documentation and set up with !include_dir_merge_named and having to keep up with different indentation than most examples that can be found in communities etc.

* Update source/_docs/configuration/packages.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_docs/configuration/packages.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_docs/configuration/packages.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_docs/configuration/packages.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update packages.markdown

added just a '  :)

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

blakadder committed:
Fix link (#27858)

cassidyjames committed:
SolarEdge: Update link to web portal (#27853)

There's no obvious way to get from the SolarEdge home page to the monitoring site, which is where you actually need to go to log in and follow the instructions for getting an API key.

kasabov committed:
Bugfix wording (#27857)

Bugfix wording around "allowing for much more flexible automations than.." (was allowing much more flexible automations that..)

ludeeus committed:
Remove note about guest access (#27844)

rccoleman committed:
Fix template in the "text" field (#27871)

ludeeus committed:
Revert "Remove note about guest access (#27844)" (#27877)

This reverts commit ee0ce1fc0d586d39f1515cf8e8df82a37304b052.

curtgrimes committed:
Add missing remote commands to Apple TV docs (#27875)

Add missing remote commands to Apple TV docs based on this list https://pyatv.dev/api/interface#pyatv.interface.RemoteControl as recommended by https://community.home-assistant.io/t/apple-tv-remote-implementation/194317/43.

Fix a couple minor typos.

Hedda committed:
Update zha.markdown (#27842)

Add URLs linking to the documentation of other integrations that device types in the ZHA integration make use of.

rsporsche committed:
Update cover.template.markdown (#27878)

c0ffeeca7 committed:
Reintroduce example that was removed before (#27876)

c0ffeeca7 committed:
M-integrations: replace term component by integration (#27795)

c0ffeeca7 committed:
Backup: small style changes (#27881)


- small style fixes in text

* Add my link

c0ffeeca7 committed:
TTS in automations: remove YAML example (#27885)

Hedda committed:
Update zha.markdown (#27884)

Try to fix URL links I broke in https://github.com/home-assistant/home-assistant.io/pull/27842

bdraco committed:
Add more information about slow bluetooth connections (#27869)

* Add more information about slow bluetooth connections

* Update source/_integrations/bluetooth.markdown

* Apply suggestions from code review

* Apply suggestions from code review

* Update source/_integrations/bluetooth.markdown

* Update source/_integrations/bluetooth.markdown

* Update source/_integrations/bluetooth.markdown

* Update source/_integrations/bluetooth.markdown

* Update source/_integrations/bluetooth.markdown

* Update source/_integrations/bluetooth.markdown

* Apply suggestions from code review

* Update source/_integrations/bluetooth.markdown

* tiny tweak

* Update source/_integrations/bluetooth.markdown

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

gpambrozio committed:
Fix spelling. (#27897)

beMoD committed:
Update Sun trigger Offset (#27896)

Update description of the offset trigger to reflect the possibility to also use number of seconds.

zlinoliver committed:
update with API services (#27901)

dependabot[bot] committed:
Bump sass-embedded from 1.63.4 to 1.63.5 (#27903)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Valloric committed:
Update editor.markdown (typo fix) (#27911)

dependabot[bot] committed:
Bump sass-embedded from 1.63.5 to 1.63.6 (#27919)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump textlint-rule-terminology from 3.0.4 to 3.0.5 (#27918)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

c0ffeeca7 committed:
Standardization: Change markup (#27921)

c0ffeeca7 committed:
Prusalink: add MK4 as supported (#27922)

jbouwh committed:
Correct device description for MQTT platforms (#27927)

rytilahti committed:
Improve issue reporting instructions (#27931)

* Improve issue reporting instructions

* Tiny tweaks

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

c0ffeeca7 committed:
Reporting issues: update (#27935)

Co-authored-by: Franck Nijhof <git@frenck.dev>

gjohansson-ST committed:
Fix documentation for Command line (#27938)

* Fix documentation Command Line

* Update source/_integrations/command_line.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Remove last section

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

frenck committed:
2023.6.3 (#27942)

gbleaney committed:
Tweak wording of assistant setup process (#27948)

Making some small tweaks for formatting consistency and to match new language in the google console.

wrt54g committed:
Update Kaiterra URL (#27959)

c0ffeeca7 committed:
Using blueprints: tiny style tweaks (#27968)

b-uwe committed:
Collaboration blog post HA // KNX (#27940)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Commit:
Update rfxtrx.markdown (#27985)

Add MHz

c0ffeeca7 committed:
Google Assistant: secure devices (#27987)

c0ffeeca7 committed:
HomeKit: add Thread device (#27892)

Co-authored-by: Stefan Agner <stefan@agner.ch>

slovdahl committed:
Fix YAML syntax error in example (#27990)

dannytsang committed:
Updated instructions to create a local user for Unifi Protect Integration (#28003)

* Updated instructions to create a local user.

Based on Unifi OS 3.0.20.

* Fixed capitalisation of "IP".

* Update source/_integrations/unifiprotect.markdown

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

marcelveldt committed:
Small improvements to the Matter documentation (#28010)

* add link to thread/docker documentation

* remove redundant text about current state

platform support is already mentioned in the top, info is outdated

* remove redundant or outdated info

* add text about running matter server in docker

* add switchbot docs

* another note about thread

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Update source/_integrations/matter.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Fix terminology issues I introduced with my review comments

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

EnthusiastDeveloper committed:
Fix typo ibeacon.markdown (#28024)

Replace "light of sight" with the correct English term

APayerl committed:
Update modbus.markdown (#28028)

Add link to light section in modbus documentation

grapesmith committed:
Add unsupported model section for TP-Link Kasa integration (#28042)

* Add unsupported model section for TP-Link Kasa integration

Ref: https://github.com/home-assistant/core/issues/94947

Some newer models of TP-Link Kasa devices are incompatible with the integration. Creating a section for them and adding one known model. Additionally, cross link to the Matter integration where it has limited support.

* tiny tweak

change link to point to link to the topic title within the matter page

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

dependabot[bot] committed:
Bump textlint from 13.3.2 to 13.3.3 (#28044)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

trickykid1982 committed:
DenonAvr add AVR-1713 as supported (#28049)

ultrafez committed:
Make clearer that all automation conditions must pass (#28052)

* Make clearer that all automation conditions must pass

Clarify that if any automation conditions fail, the automation actions will not be executed.

* tiny tweaks

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

emontnemery committed:
Remove documentation of deprecated vacuum services (#28054)

mehuman committed:
Troubleshooting section no longer needed after move to new library (#28053)

emontnemery committed:
Remove documentation of deprecated MQTT vacuum protocol (#28056)

emontnemery committed:
Add documentation of deprecated vacuum services to ecovacs (#28055)

tronikos committed:
Merge branch 'home-assistant:current' into patch-1
